src/main/scala/mesosphere/marathon/core/instance/Reservation.scala

```diff
@@ -12,6 +12,34 @@ import play.api.libs.json._
 case class Reservation(volumeIds: Seq[LocalVolumeId], state: Reservation.State)
 
 object Reservation {
+
+  /**
+    * Reservation.Id represents a reservation ID which is used in a task reservation labels.
+    */
+  case class Id(runSpecId: String, separator: String, prefix: Option[String], uuid: String) {
+    override def toString: String = prefix match {
+      case Some(prefixStr) => runSpecId + separator + prefixStr + uuid
+      case None => runSpecId + separator + uuid
+    }
+
+    def instanceId: Instance.Id = prefix match {
+      case Some(prefixStr) => Instance.Id(runSpecId + separator + prefixStr + uuid)
```

Comment:
I don't think that reservation id class should have such a deep knowledge of how instance id is concatenated from various string. I think this should definitely be somewhere near Instance.Id implementation

Reply:
It is implied by the definition of `Instance.Id`.
```
case class Id(idString: String) extends Ordered[Id]
```
It is scattered all over the code. Though I agree, it probably warrants a refactoring.